utils/build/build.mk

```diff
@@ -1,25 +1,17 @@
 # This makefile configures variables related to compilation, e.g. compiler flags.
 # Furthermore, it defines compilation recipes
-CFLAGS   := -pthread -Wall
-LFLAGS   := -lpthread -pthread
+CFLAGS   := $(CFLAGS) -pthread -Wall
```

Comment:
The problem was that since the main `Makefile` includes both `config.mk` and `build.mk`, and the flag to use c++11 is added in `config.mk` and `config.mk` gets included before `build.mk`, the `:=` operator in `build.mk` would simply overwrite the c++11 flag and would cause the compilation to fail.